plugins/amdgpu/amdgpu_plugin.c

```diff
@@ -621,6 +626,83 @@ void *dump_bo_contents(void *_thread_data)
         return NULL;
 };
 
+int check_hsakmt_shared_mem(uint64_t *shared_mem_size, uint32_t *shared_mem_magic)
+{
+	int ret, fd;
+	struct stat st;
+	uint32_t *shared_mem_ptr;
+
+	ret = stat(HSAKMT_SHM_PATH, &st);
+	if (ret) {
+		*shared_mem_size = 0;
+		return 0;
+	}
+
+	fd = shm_open(HSAKMT_SHM, O_RDONLY, 0);
+	if (fd < 0) {
+		*shared_mem_size = 0;
+		return 0;
+	}
+
+	shared_mem_ptr = mmap(0, st.st_size, PROT_READ, MAP_SHARED, fd, 0);
+	if (shared_mem_ptr == MAP_FAILED) {
+		pr_perror("amdgpu_plugin: Failed to mmap shared mem %s\n", HSAKMT_SHM);
+		return -EACCES;
+	}
+
+	*shared_mem_size = st.st_size;
+
+	/* First 4 bytes of shared file is the magic */
+	memcpy(shared_mem_magic, shared_mem_ptr, sizeof(*shared_mem_magic));
+	plugin_log_msg("amdgpu_plugin: Shared mem magic:0x%x\n", *shared_mem_magic);
+
+	close(fd);
+	return 0;
+}
+
+int restore_hsakmt_shared_mem(const uint64_t shared_mem_size, const uint32_t shared_mem_magic)
+{
+	int ret, fd;
+	struct stat st;
+	uint32_t *shared_mem_ptr;
+	sem_t *sem = SEM_FAILED;
+
+	if (!shared_mem_size)
+		return 0;
+
+	if (!stat(HSAKMT_SHM_PATH, &st)) {
+		pr_debug("amdgpu_plugin: %s already exists\n", HSAKMT_SHM_PATH);
```

Comment:
I think you forgot to change this one to plugin_log_msg

Reply:
For this one, it would be better to print if the user calls criu with -vvv